File: java/src/main/java/com/unialfa/model/Empresa.java
package com.unialfa.model;

public class Empresa {
    private int id;
    private String nome;
    private String cnpj;
    private String email;
    private String senha;
    private String cidade;
    private String status;

    public Empresa() {}

    public Empresa(String nome, String cnpj, String email, String senha, String cidade, String status) {
        this.nome = nome;
        this.cnpj = cnpj;
        this.email = email;
        this.senha = senha;
        this.cidade = cidade;
        this.status = status;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getCnpj() {
        return cnpj;
    }

    public void setCnpj(String cnpj) {
        this.cnpj = cnpj;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getSenha() {
        return senha;
    }

    public void setSenha(String senha) {
        this.senha = senha;
    }

    public String getCidade() {
        return cidade;
    }

    public void setCidade(String cidade) {
        this.cidade = cidade;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getResumo() {
        return "Empresa{" +
                "id=" + id +
                ", nome='" + nome + '\'' +
                ", cnpj='" + cnpj + '\'' +
                ", email='" + email + '\'' +
                ", cidade='" + cidade + '\'' +
                ", status='" + status + '\'' +
                '}';
    }

    @Override
    public String toString() {
        return getResumo();
    }
}

File: php/classes/Empresa.php
<?php

require_once __DIR__ . '/Usuario.php';

class Empresa extends Usuario {
    private string $cnpj;
    private string $telefone;
    private string $cidade;
    private string $status; // pendente | ativa | bloqueada

    public function __construct(
        int    $id,
        string $nome,
        string $email,
        string $cnpj     = '',
        string $telefone = '',
        string $cidade   = '',
        string $status   = 'pendente',
        string $senha    = ''
    ) {
        parent::__construct($id, $nome, $email, $senha);
        $this->cnpj     = $cnpj;
        $this->telefone = $telefone;
        $this->cidade   = $cidade;
        $this->status   = $status;
    }

    public function getCnpj():     string { return $this->cnpj; }
    public function getTelefone(): string { return $this->telefone; }
    public function getCidade():   string { return $this->cidade; }
    public function getStatus():   string { return $this->status; }

    public function isAtiva(): bool { return $this->status === 'ativa'; }

    public function getPerfil(): string {
        return "Empresa — {$this->nome} | CNPJ: {$this->cnpj} | {$this->cidade}";
    }

    public function getTipo(): string { return 'empresa'; }

    public static function fromArray(array $data): self {
        return new self(
            $data['id']       ?? 0,
            $data['nome']     ?? '',
            $data['email']    ?? '',
            $data['cnpj']     ?? '',
            $data['telefone'] ?? '',
            $data['cidade']   ?? '',
            $data['status']   ?? 'pendente'
        );
    }
}